plugins/cache/cache_main.ml

```diff
@@ -167,6 +189,31 @@ module Index = struct
           let entry,res = f entry in
           update_entry idx src entry,res)
 
+  let upgrade_old_index file version =
+    let () = match version with
+      | 1 ->
+        let old =
+          try Sexp.load_sexp file |> index_of_sexp
+          with _ -> empty in
+        index_to_file (cache_dir () / index_file) old;
+      | x ->
+        warning
+          "can't update index version from %d to %d" x index_version in
+    Sys.remove file
+
+  let get_index () =
+    FileUtil.ls (cache_dir ()) |> List.find ~f:is_index
+
+  let verify () =
```

Comment:
the name is misleading, you're not verifying anything, you are actually upgrading it, if needed.